### PR TITLE
Scheduled FIRRTL output format; minor refactoring, config and codestyle fixes

### PR DESCRIPTION
**Pull request by Muxianesty**
This Pull Request introduces a scheduling FIRRTL output format (described in #26) as well as a couple of minor configuration and codestyle fixes.

### CODE REVIEW

**Review by ssmolov — COMMENTED** [review]
Tests (unit-level or system-level) are requested here to illustrate scenarios of existing cmdline options usage.

**Comment by Muxianesty** [comment]
> Tests (unit-level or system-level) are requested here to illustrate scenarios of existing cmdline options usage.

In this Pull Request only a FIRRTL output format was added. In the following Pull Requests new tests will be added, and system tests will be added at the CI-level (#42).

File: src/model/dfcxx/lib/dfcxx/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,17 @@ add_subdirectory(typebuilders)
 add_subdirectory(vars)
 add_subdirectory(varbuilders)
 
+## By default LLVM, MLIR and DFCIR are compiled without RTTI, but not DFCxx.
+## So using core Pass functionality from MLIR in converter.cpp causes
+## linker-level errors ("undefined reference to `typeinfo for mlir::Pass'").
+## The following lines explicitly disable RTTI for DFCxx.
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
+  set(DISABLE_RTTI_FLAG "/GR-")
+else()
```
[the remark this answers is not in the document]
Comment (Muxianesty) [reply src/model/dfcxx/lib/dfcxx/CMakeLists.txt]: Fixed.

**Review by ssmolov — APPROVED** [review]
ok

**Pull request merged**